Demos/Cube/Vec3f.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2020, Stephan Unverwerth <s.unverwerth@gmx.de>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+struct Vec3f {
```

Comment:
This looks out of place. A more "Serenity" name might be `FloatVector3`, `Vector3<float>` or just `Vector3`.
Or maybe this is a term of art too?

Regardless, we tend to prefer `class` interfaces with getters and setters for complex types like this.

Reply:
I agree about the naming. I guess you can always `typedef` it if you like to type less.
I disagree about the getters/setters. These are plain value types with no invariants to uphold, adding setters/getters would just make the resulting code needlessly complex and verbose. Also since these are mathematical constructs they are unlikely to change in the future, so an access to x should always return the member x, nothing else.

Reply:
Sure, but it's a stylistic choice for consistency over perceived simplicity. Look at all the other LibGfx primitives like `Point`, `Rect`, `Size`, etc. :)

Reply:
Ok, this is a Serenity contribution after all, and not my personal project :^)